src/allreduce_base.h

```diff
@@ -546,6 +546,10 @@ class AllreduceBase : public IEngine {
   int rabit_bootstrap_cache = 0;
   // enable detailed logging
   int rabit_debug = 0;
+  // by default, if rabit worker not recover in half an hour exit
+  int timeout_sec = 1800;
+  // flag to enable rabit_timeout
+  int rabit_timeout = 0;
```

Comment:
If it's a flag, then it should be a enum class or bits with documented finite values.  In `all_reduce_base.cc` you write its value with `atoi`, I'm gonna have to assume this is not a simple flag.

Reply:
fixed and using bool type, user can set =1 or =true